src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -143,6 +143,9 @@ class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExte
       case Some(_) =>
         for {
           _ <- tosService.acceptTosStatus(userId)
+          _ <- directoryDAO.isEnabled(userId, samRequestContext).flatMap(enabled => {
+            if (enabled) registrationDAO.enableIdentity(userId, samRequestContext) else IO.none
+          })
           status <- IO.fromFuture(IO(getUserStatus(userId, false, samRequestContext)))
         } yield status
```

Comment:
for-comprehensions are implemented under the covers as flatMaps, so you could write this as:
```
        for {
          _ <- tosService.acceptTosStatus(userId)
          directoryEnabled <- directoryDAO.isEnabled(userId, samRequestContext)
          _ <- if (directoryEnabled) registrationDAO.enableIdentity(userId, samRequestContext) else IO.none
          status <- IO.fromFuture(IO(getUserStatus(userId, false, samRequestContext)))
        } yield status
```
I think both approaches have their merits, I don't feel strongly about changing this.

Reply:
Ah! I prefer your approach given that we're already using for-iterations- updating-